### PR TITLE
chore: update KSQL metrics context labels

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 

- lower case the resource type
- adds the following configs to ksqlConfig when initializing KSQL for either headless or interactive mode so that all components get the same base set of labels
```
metrics.context.resource.kafka.cluster.id
metrics.context.resource.type
metrics.context.resource.cluster.id
metrics.context.resource.version
metrics.context.resource.commit.id
```
### Testing done 
Rechecking labels present in the staging URL for Druid

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java [hunk 1]
```diff
@@ -143,22 +142,16 @@ public static void addConfigurableReporter(
   }
 
   public static Map<String, Object> addConfluentMetricsContextConfigs(
-      final String ksqlServiceId
+      final String ksqlServiceId,
+      final String kafkaClusterId
   ) {
     final Map<String, Object> updatedProps = new HashMap<>();
     updatedProps.put(RESOURCE_LABEL_TYPE, KSQL_RESOURCE_TYPE);
     updatedProps.put(RESOURCE_LABEL_CLUSTER_ID, ksqlServiceId);
```
Comment (xiaodongdu) [inline ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:150]: If ksql.cluster.id is the same as ksql.service.id, why do we need both?

File: ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java [hunk 2]
```diff
@@ -143,22 +142,16 @@ public static void addConfigurableReporter(
...
-    return updatedProps;
-  }
-
-  public static Map<String, Object> addConfluentMetricsContextConfigsForKsql(
-      final String ksqlServiceId
-  ) {
-    final Map<String, Object> updatedProps = new HashMap<>();
     updatedProps.put(RESOURCE_LABEL_KSQL_SERVICE_ID, ksqlServiceId);
+    updatedProps.put(RESOURCE_LABEL_KAFKA_CLUSTER_ID, kafkaClusterId);
```
Comment (xiaodongdu) [inline ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:152]: if ksql.cluster.id and ksql.service.id are the same, why do we need both?

**Comment by xiaodongdu** [comment]
Fix looks fine, one thing I'm not clear, metrics.context.resource.cluster.id and metrics.context.resource.service.id are both populated with service id, why do we need both?

File: ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java (hunk 2 above)
Comment (stevenpyzhang) [reply ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:149]: we can remove RESOURCE_LABEL_KSQL_SERVICE_ID and just keep the cluster id label. The service id of a KSQL server is the only id for it so it essentially is the cluster id

Comment (xiaodongdu) [reply ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:149]: RESOURCE_LABEL_CLUSTER_ID and RESOURCE_LABEL_KAFKA_CLUSTER_ID are the same. I would recommend to keep RESOURCE_LABEL_KSQL_SERVICE_ID which is more meaningful for KSQL.

Comment (stevenpyzhang) [reply ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:149]: Oh I see, I was under the impression that the cluster id meant the ksql cluster id, not the kafka one. If that's the case, then I'll keep RESOURCE_LABEL_KSQL_SERVICE_ID

Comment (xvrl) [reply ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:149]: @stevenpyzhang @xiaodongdu I think it makes more sense to keep `ksql.cluster.id` as the KSQL service ID, and `ksql.kafka.cluster.id` as the backing cluster id to align with the terminology we already use in RBAC https://docs.confluent.io/current/security/rbac/rbac-get-cluster-ids.html#ksqldb-example

We should fix the other components if they don't align.

Comment (stevenpyzhang) [reply ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricCollectors.java:149]: @xvrl I've updated the PR to match with this ^

**Pull request merged**